metaflow/client/core.py

```diff
@@ -1509,10 +1506,24 @@ def control_tasks(self, *tags):
             Iterator over Control Task objects in this step
         """
         children = super(Step, self).__iter__()
-        filter_tags = [CONTROL_TASK_TAG]
-        filter_tags.extend(tags)
         for child in children:
-            if all(tag in child.tags for tag in filter_tags):
+            # first filter by standard tag filters
+            if not all(tag in child.tags for tag in tags):
+                continue
+            # then look for control task indicator in one of two ways
+            is_control_task = False
+            # Look in tags - this path will activate for metadata service backends that pre-date tag mutation release
```

Comment:
nit: these comment lines look quite long. Not sure why black isn't shortening them.